book/src/project-policies.md

```diff
@@ -0,0 +1,61 @@
+# Versioning, Support, and Kubernetes Compatibility
+
+## Versioning
+
+Each Kubernetes CSI component version is expressed as *x.y.z*, where *x* is the
+major version, *y* is the minor version, and *z* is the patch verseion,
+following [Semantic Versioning](https://semver.org/).
+
+Patch version releases only contain bug fixes that do not break any backwards
+compatibility.
+
+Minor version releases may contain new functionality that do not
+break any backwards compatibility (except for alpha features).
+
+Major version releases may contain new functionality or fixes that may break
+backwards compatibility with previous major releases. Changes that require a
+major version increase include: removing or changing API/flags//behavior, new
+RBAC requirements that are not opt-in, new Kubernetes minimum version
+requirements.
+
+## Support
+
+The Kubernetes CSI project follows the broader Kubernetes project on support.
+Every minor release branch will be supported with patch releases for 1 year.
```

Comment:
"... will be supported with patch releases *at least* for one year, starting with the first release of that branch."

If there's no newer release branch to replace it, then the latest one will continue to be supported for more than one year.

Given that the revised criteria is now strictly time based, would it make sense to add a column with that release date to the version tables of each sidecar?

For example, when looking at https://kubernetes-csi.github.io/docs/external-snapshotter.html it is not obvious which of the branches are recent enough to still be supported.

Reply:
+1 for listing expected end of support.

> If there's no newer release branch to replace it, then the latest one will continue to be supported for more than one year.

This implies we can stop supporting older branches that were released 1 year ago at the point we release a new major version. IMO, we should have some overlap where both the old branch + the new branch are supported, to give driver or Kubernetes vendors time to update everything. With my storage vendor hat on, it should be at least 3 months, preferably even more.

Reply:
tried to word it in the least confusing way... ptal

Reply:
Regarding updating the sidecar tables, I agree we should do that and also add the date of first release. But I want to do it in a separate PR, it's going to take a fair amount of time to look everything up.